build/helper/metadata_add_all.py

```diff
@@ -271,7 +273,10 @@ def _add_render_in_session_base(f):
 def _add_is_repeated_capability(parameter):
     '''Adds a boolean 'is_repeated_capability' to the parameter metadata by inferring it from its name, if not previously populated.'''
     if 'is_repeated_capability' not in parameter:
-        if parameter['name'] in _repeated_capability_parameter_names:
+        param_name = parameter['name']
+        if 'python_name_override' in parameter and parameter['python_name_override']:
```

Comment:
I see few problems with this (but I like the idea of fixing the issue):
1) By design the expanded metadata should contain all the keys in all the instances but `python_name_override` is only added to some
2) `_repeated_capability_parameter_names`  list currently contains camelCase-formatted strings. But `python_name` will always be snake_case. The current code will not work when `python_name_override` is true and we want rep cap tip added based on `python_name`
3) We can infer that `python_name` is overriding `name` by comparing the two. Do we really need `python_name_override`?

I think this change is non-trivial enough that it should be done in a separate PR.

Reply:
1. This is a valid point, but I think it would be straightforward enough for me to always add `python_name_override` (either true or false)
2. That's a very good point. I could address this by **a)** adding underscored versions of the parameter names to `_repeated_capability_parameter_names`, or **b)** creating a separate collection of underscored names just for python, or **c)** writing a `snakecase_to_camelcase` helper to modify the python name before looking for it in `_repeated_capability_parameter_names`.
3. The reason for the override flag is because in `_add_python_method_name` we set `python_name `to match the regular name. So that by the time we're in def `_add_is_repeated_capability`, we've lost that distinction: we won't know if we're looking at the `python_name` that came from hapigen, or the one that was 
set in `_add_is_repeated_capability`.

My motivation for fixing it with this PR is because this PR was the first to include support for the `python_name` hapigen flag for function parameters. So it made sense to me to address the problem in the same PR. But if you don't agree with how I'm planning to fix items 1 and 2, or if you think that there might be additional complications beyond the ones you listed, let me know and I'll be happy to create an issue and revert this change. Let me know what you think.

Reply:
I've pushed some changes that address the first two items above. But like I said before, if you're still uncomfortable with this change let me know and I'll revert it.

Reply:
Your changes address my concerns, let's keep it.

Reply:
Not relevant anymore, resolving the conversation.